src/dask_awkward/lib/structure.py

```diff
@@ -498,9 +498,47 @@ def values_astype(array, to, highlevel=True, behavior=None):
     raise DaskAwkwardNotImplemented("TODO")
 
 
+class _WhereFn:
+    def __init__(
+        self,
+        mergebool: bool = True,
+        highlevel: bool = True,
+        behavior: dict | None = None,
+    ) -> None:
+        self.mergebool = mergebool
+        self.highlevel = highlevel
+        self.behavior = behavior
+
+    def __call__(self, condition: ak.Array, x: ak.Array, y: ak.Array) -> ak.Array:
+        return ak.where(
+            condition,
+            x,
+            y,
+            mergebool=self.mergebool,
+            highlevel=self.highlevel,
+            behavior=self.behavior,
+        )
+
+
 @borrow_docstring(ak.where)
-def where(condition, *args, **kwargs):
-    raise DaskAwkwardNotImplemented("TODO")
+def where(
+    condition: Array,
+    x: Array,
+    y: Array,
+    mergebool: bool = True,
+    highlevel: bool = True,
+    behavior: dict | None = None,
+) -> Array:
+    if not highlevel:
+        raise ValueError("Only highlevel=True is supported")
+    return map_partitions(
```

Comment:
since `ak.where` requires `condition` `x` and `y` to have equal lengths we should add a partition/divisions check (you may need to import `dask.lib.core.comaptible_partitions` and `dask.utils.IncompatiblePartitions`)
```suggestion
    if not compatible_partitions(condition, x, y):
        raise IncompatiblePartitions("where", condition, x, y)
    return map_partitions(
```

Reply:
Thanks! I was gonna launch a few more feature addition PRs and then I'll get to the review here.